### PR TITLE
canPerformAction updated to allow pasting images into TextView or into Plugins

### PR DESCRIPTION
**Pull request by eliburke**
* canPerformAction updated to explicitly allow image paste if isImagePasteEnabled or if any inputPlugins are available, or defer to system implementation
* paste updated to try image paste first, fall through to system paste behavior
* InputTextView source comments updated to document new behavior

### CODE REVIEW

File: Sources/Views/InputTextView.swift [hunk 1]
```diff
@@ -232,25 +233,28 @@ open class InputTextView: UITextView {
...
+                return
+            } else {
+                for plugin in inputBarAccessoryView?.inputPlugins ?? [] {
+                    if plugin.handleInput(of: image) {
+                        return
+                    }
                 }
             }
         }
+        return super.paste(sender)
```
Comment (austinwright) [inline Sources/Views/InputTextView.swift:257]: Is the intention here to execute `return super.paste(sender)` if there are no plugins, or no plugins return `true` from `handleInput(of: image)`?

**Comment by Kaspik** [comment]
@eliburke Any update here?

**Comment by Kaspik** [comment]
This PR has been open too long. Waiting for the changes based on the review.
Will close if there is no further activity in upcoming days. :)

**Comment by eliburke** [comment]
Apologies, I missed the request for clarification back in 2020. The idea for Sources/Views/InputTextView.swift and the `return super.paste(sender)` line was to do both: call the superclass implementation if there are no plugins, OR if there are plugins but none handle the image. 

If i recall, the InputBarAccessoryView had been changed in a way that preventing image paste from working correctly in at least one case, based on functions that were written and commit history. This was supposed to fix it. Unfortunately I haven't touched my project in two years, so I would not feel comfortable making further edits to the PR.

**Comment by Kaspik** [comment]
Thanks for the answer! I'll close the PR for now but if ever got back to it, feel free to reopen with the updated code based on review above. :)

**Pull request closed without merging**